lib/utils.ts

```diff
@@ -40,6 +40,20 @@ export function escapeRegex(str: string) {
 	return str.replace(/[\\^$.*+?()[\]{}|]/g, '\\$&');
 }
 
+export function listMethods(obj: any, checker?: (prop: string, val: any) => boolean) {
+	const methods = new Set<string>();
+	let current = obj;
+	do {
+		const curProps = Object.getOwnPropertyNames(current);
+		for (const prop of curProps) {
+			if (typeof obj[prop] !== 'function') continue;
+			if (checker && !checker(prop, obj[prop])) continue;
+			methods.add(prop);
+		}
+	} while ((current = Object.getPrototypeOf(current)));
+	return [...methods.keys()];
+}
+
```

Comment:
I don't think this belongs in Utils. It's very specific to PS's hotpatching system, `chat-commands/admin` definitely strikes me as the correct place. Passing in an external checker also seems to be unnecessary abstraction (lower readability for no gain).